homeassistant/components/backup/util.py

```diff
@@ -288,6 +309,156 @@ def _decrypt_backup(
             output_tar.addfile(decrypted_obj, decrypted)
 
 
+def encrypt_backup(
```

Comment:
I think we should add unit tests for the new util functions. This is becoming hard to follow if you're not very familiar with all the details.

Reply:
OK, I'll start with unit tests, and then add end-to-end tests as a final step?